### ISSUE TITLE
no-unused-variable: false positive for exported function return type

### ISSUE THREAD

**Issue by msklvsk** [palantir/tslint#1157]
### Bug Report
- **TSLint version**: `3.8.0`
- **TypeScript version**: `1.8.10`
- **Running TSLint via**: gulp-tslint
#### TypeScript code being linted

``` ts
import {Server, createServer} from 'http';
// unused variable: 'Server' (while it is used as return type)

export function getServer() {
  return createServer();
}
```

with `tslint.json`:

``` json
{
  "rules": {
    "no-unused-variable": true
  }
}
```
#### Actual behavior

An error _unused variable: 'Server'_ is reported, but it is used as a return type and removing `Server` from imports causes a native typescript error: _Return type of exported function has or is using name 'Server' from external module "http" but cannot be named_
#### Expected behavior

no tslint error is reported

**Comment by adidahiya** [comment]
Hm, this is an interesting problem with implicit typedefs that hasn't really come up before. Not sure if we'll be able to identify `Server` as an implicitly-used type, especially without full type checker info (#680). For now I would suggest adding an explicit return type to `getServer` (IMO it's good practice to do so for exported APIs anyway).

**Comment by andy-hanson** [comment]
This is handled in #2235. Although it *should* be fixed in TypeScript itself, for which the issue is Microsoft/TypeScript#9944 or Microsoft/TypeScript#13694.
